cherab/generomak/plasma/plasma.py

```diff
@@ -215,3 +217,332 @@ def get_edge_plasma(atomic_data=None, parent=None, name="Generomak edge plasma")
+                                                                              nimp_core, nimp_convexity, nimp_concavity, xmin=1, xmax=0) 
+    profiles["composition"]["carbon"][6]["f1d_vtor"] = f1d_vtor
+    profiles["composition"]["carbon"][6]["f1d_vpol"] = f1d_vpol
+    profiles["composition"]["carbon"][6]["f1d_vnorm"] = f1d_vnorm
+
+    # setup CX+ profile shapes with temperature as double parabolas and density with decaying exponentials
+    for chrg in range(6):
+        profiles["composition"]["carbon"][chrg]["f1d_temperature"] = get_double_parabola(lcfs_values["composition"]["carbon"][chrg]["temperature"],
+                                                                                         timp_core, timp_convexity, timp_concavity, xmin=1, xmax=0)
+        profiles["composition"]["carbon"][chrg]["f1d_density"] = get_exponential_growth(lcfs_values["composition"]["carbon"][chrg]["density"], nimp_decay)
+        profiles["composition"]["carbon"][chrg]["f1d_vtor"] = f1d_vtor
+        profiles["composition"]["carbon"][chrg]["f1d_vpol"] = f1d_vpol
+        profiles["composition"]["carbon"][chrg]["f1d_vnorm"] = f1d_vnorm
+
+    return profiles.freeze()
+
+def get_core_distributions(profiles=None, equilibrium=None):
+    """
+    Returns a dictionary of core plasma species Maxwellian distributions.
+
+    :param profiles: Dictionary of core particle profiles.  The dictionary has to have the same form 
+                     as the one returned by the function get_core_profiles_description. 
+                     The default value is the value returned by the call get_core_profiles_description().
+    :param equilibrium: an instance of EFITEquilibrium.
+    :return:  dictionary of core plasma species with Maxwellian distribution
+    """
+    # get core profile data if not passed sa argument
+    if profiles is None:
+        profiles = get_core_profiles_description()
+    
+    # load plasma equilibrium if not passed as argument
+    if equilibrium is None:
+        equilibrium = load_equilibrium()
+
+
+    # build a dictionary with Maxwellian distributions
+    species = RecursiveDict()
+    species["electron"] = get_maxwellian_distribution(equilibrium, rest_mass=electron_mass,
+                                                            **profiles["electron"])
+    for name, spec in profiles["composition"].items():
+        spec_cherab = _get_cherab_element(name)
+        for chrg, desc in spec.items():
+            rest_mass = atomic_mass * spec_cherab.atomic_weight
+            species["composition"][name][chrg] = get_maxwellian_distribution(equilibrium, rest_mass=rest_mass,  **desc)
+        
+    return species.freeze()
+
+def get_core_plasma(distributions=None, atomic_data=None, parent=None, name="Generomak core plasma"):
+    """
+    Provides Generomak core plasma.
+
+    :param distributions: A dictionary of plasma distributions. Has to have the same for mas the
```

Comment:
`Has to have the same format as the ...`

Reply:
corrected